src/tool.rs

```diff
@@ -134,6 +134,9 @@ impl Tool {
                 })?;
             tmp.file()
                 .write_all(include_bytes!("detect_compiler_family.c"))?;
+            // Close the file handle *now*, otherwise the compiler may fail to open it on Windows
+            // (#1082). The file stays on disk and its path remains valid until `tmp` is dropped.
+            tmp.close();
```

Comment:
Let's also do a `tmp.flush()?;`

Reply:
That would call the `Write::flush` method, which for `File` is a no-op on all mainstream platforms including all that cc supports right now: [Windows](https://github.com/rust-lang/rust/blob/16e8803579e5914a17a4924f47faaefc040a982d/library/std/src/sys/pal/windows/fs.rs#L486), [Unix](https://github.com/rust-lang/rust/blob/16e8803579e5914a17a4924f47faaefc040a982d/library/std/src/sys/pal/unix/fs.rs#L1294), and [wasi](https://github.com/rust-lang/rust/blob/16e8803579e5914a17a4924f47faaefc040a982d/library/std/src/sys/pal/wasi/fs.rs#L460). If you intend the call as precaution against adding a `BufWriter` or something in the future, sure, I can add it. However, if you want to surface I/O errors that would otherwise be detected while closing the file and ignored by dropping, we'll need `File::sync_all` as discussed in `File`'s documentation. Which option do you prefer?

Reply:
Thanks for investigating this, I'd like `file.flush()?;` and `file.sync_data()?;` to be called.

I don't think we need `file.sync_metadata()?;`, since it's a tempfile.

Reply:
Done.